GitUI/HelperDialogs/FormResetBranch.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using GitCommands;
+using GitExtUtils.GitUI.Theming;
+using ResourceManager;
+
+namespace GitUI.HelperDialogs
+{
+    public partial class FormResetBranch : GitModuleForm
+    {
+        private readonly IReadOnlyList<GitRef> _gitReferences;
+
+        public FormResetBranch(GitUICommands gitUiCommands, GitRevision revision, IReadOnlyList<GitRef> gitReferences)
+            : base(gitUiCommands)
+        {
+            Revision = revision;
+
+            InitializeComponent();
+            InitializeComplete();
+
+            _gitReferences = gitReferences;
+            if (_gitReferences != null)
+            {
+                Branches.DisplayMember = nameof(GitRef.Name);
+                Branches.Items.AddRange(_gitReferences.Where(h => h.IsHead && !h.IsRemote).ToArray());
+            }
+
+            labelResetBranchWarning.SetForeColorForBackColor();
+        }
+
+        public GitRevision Revision { get; set; }
+        public GitRef GitReference { get; set; }
+
+        private void FormResetBranch_Load(object sender, EventArgs e)
+        {
+            if (Revision == null)
+            {
+                throw new Exception("No revision");
+            }
+
+            commitSummaryUserControl.Revision = Revision;
+
+            if (Branches.Text.Length == 0)
+            {
+                Branches.DroppedDown = true;
```

Comment:
I prefer to not pull down the branches by default, it seems like unintentional to get the list in your face.
Resetting any branch is not done very often, it should not be a problem to click once more

Reply:
> I prefer to not pull down the branches by default.

A branch must be selected. What do the GUI designers say?

Reply:
>What do the GUI designers say?

I'm absolutely not a GUI designers but 😃 ...

* the display when the dropdown is opening is a little strange. The list is displayed before the rest of the form...
* all the form could be totally hidden by the branch list so, on the first use, the user could have difficulties to understand how the form works
* but once it is used one time, I find it mostly good.

So, even if it is not a full win, that's interesting.
But once again, not a strong point of view for either of the solution chosen...

Reply:
I would keep the box empty and let the user press the arrow to see the branches.
No change now, I keep my comment